compiler/rustc_error_codes/src/error_codes/E0495.md

```diff
@@ -1,40 +1,36 @@
-#### Note: this error code is no longer emitted by the compiler.
+A trait implementation returns a reference without an explicit lifetime linking it to `self`.
```

Comment:
I don't think this is what this error stands for, I think this is a much more generic lifetime error from searching for the code in `./tests`.

I would probably just keep the old documentation, it's not amazing but I don't think it's possible to write documentation with great examples for this error code, it just seems like this is a very generic error code emitted in all sorts of cases.

@compiler-errors maybe you know more about this error code and whether what I said is correct.